atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -297,9 +308,16 @@ public synchronized void shutdown() {
             return;
         }
         log.info("Signalling background sweeper to shut down.");
+        // Interrupt the daemon, whatever lock it may be waiting on.
         daemon.interrupt();
+        // Ensure we do not accidentally abort shutdown if any code incorrectly swallows InterruptedExceptions
+        // on the daemon thread.
+        shuttingDown.countDown();
         try {
-            daemon.join();
+            daemon.join(MAX_DAEMON_CLEAN_SHUTDOWN_TIME_MILLIS);
+            if (daemon.isAlive()) {
+                log.error("Background sweep thread failed to shut down");
```

Comment:
Discussed offline - this might seem a bit harsh, though it definitively indicates a product bug (i.e. someone's swallowing an interrupt and slept for more than 10s)